sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/rules.scala

```diff
@@ -252,6 +251,8 @@ case class PreprocessTableInsertion(conf: SQLConf) extends Rule[LogicalPlan] {
         if (expected.dataType.sameType(actual.dataType) && expected.name == actual.name) {
           actual
         } else {
+          // Renaming is needed for handling the INSERT statement like
+          // INSERT INTO TABLE tab1 SELECT 1, 2
           Alias(Cast(actual, expected.dataType), expected.name)()
```

Comment:
Submitted another PR: https://github.com/apache/spark/pull/15615. Could you check which one is better? Thank you!